lib/view/scan_view.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:flutter/material.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+
+class ScanView extends StatefulWidget {
+  const ScanView({Key? key}) : super(key: key);
+
+  @override
+  _ScanViewState createState() => _ScanViewState();
+}
+
+class _ScanViewState extends State<ScanView> {
+  final GlobalKey qrKey = GlobalKey(debugLabel: 'QR');
+  Barcode? result;
```

Comment:
**question:** Why use state variables here when you use Providers elsewhere for state management?